dapr/clients/grpc/_response.py

```diff
@@ -5,54 +5,64 @@
 Licensed under the MIT License.
 """
 
-from typing import Optional
-from typing import Dict
+from typing import Dict, Optional, Union
 
 from google.protobuf.any_pb2 import Any as GrpcAny
 from google.protobuf.message import Message as GrpcMessage
 
-from dapr.clients.grpc._helpers import MetadataDict, MetadataTuple
+from dapr.clients.base import DEFAULT_JSON_CONTENT_TYPE
+from dapr.clients.grpc._helpers import (
+    MetadataDict,
+    MetadataTuple,
+    to_bytes,
+    to_str,
+    tuple_to_dict,
+    unpack,
+)
 
 
 class DaprResponse:
     """A base class for Dapr Response.
 
-    This is the base class for Dapr Response. User can gets the headers and trailers as
-    a dict.
+    This is the base class for Dapr Response. User can get the headers as a dict.
 
     Attributes:
         headers(dict): A dict to include the headers from Dapr gRPC Response.
-        trailers(dict): A dict to include the trailers from Dapr gRPC Response.
     """
 
     def __init__(
             self,
-            headers: Optional[MetadataTuple] = (),
-            trailers: Optional[MetadataTuple] = ()):
+            headers: MetadataTuple = ()):
         """Inits DapResponse with headers and trailers.
 
         Args:
-            headers (Tuple, optional): the tuple for the headers from gRPC response
-            trailers (Tuple, optional): the tuple for the trailers from gRPC response
+            headers (tuple, optional): the tuple for the headers from response.
         """
-        self._headers = self._from_tuple(headers)
-        self._trailers = self._from_tuple(trailers)
-
-    def _from_tuple(self, metadata: Optional[MetadataTuple]) -> MetadataDict:
-        d: MetadataDict = {}
-        for k, v in metadata:  # type: ignore
-            d.setdefault(k, []).append(v)
-        return d
+        self.headers = headers  # type: ignore
 
     @property
     def headers(self) -> MetadataDict:
-        """Returns headers tuple as a dict."""
-        return self._headers
+        """Gets headers as a dict."""
+        return self.get_headers(as_dict=True)  # type: ignore
 
-    @property
-    def trailers(self) -> MetadataDict:
-        """Returns trailers tuple as a dict."""
-        return self._trailers
+    @headers.setter
+    def headers(self, val: MetadataTuple) -> None:
+        """Sets response headers."""
+        self._headers = val
+
+    def get_headers(self, as_dict: bool = False) -> Union[MetadataDict, MetadataTuple]:
+        """Gets headers from the response.
+
+        Args:
+            as_dict (bool): dict type headers if as_dict is True. Otherwise, return
+                tuple headers.
+
+        Returns:
+            dict or tuple: response headers.
+        """
+        if as_dict:
+            return tuple_to_dict(self._headers)
+        return self._headers
 
 
 class InvokeServiceResponse(DaprResponse):
```

Comment:
Why is this not `ServiceResponse` class? Wouldn't if be consistent to have Invoke in both request and response if possible or in neither?